core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -0,0 +1,169 @@
+    }
+
+    for {
+      workspace <- traceWithParent("withNewMcWorkspaceContext", parentSpan)(
+        opSpan => dataSource.inTransactionWithAttrTempTable(Set(AttributeTempTableType.Workspace))({ dataAccess =>
+          createWorkspace(workspaceRequest, dataAccess, opSpan) { createdWorkspace =>
+            DBIO.successful(createdWorkspace)
+          }
+        }, TransactionIsolation.ReadCommitted)) // read committed to avoid deadlocks on workspace attr scratch table
+    } yield workspace
+
+  }
+
+  private def createWorkspace[T](workspaceRequest: MultiCloudWorkspaceRequest, dataAccess: DataAccess, parentSpan: Span)
+                                (op: (Workspace) => ReadWriteAction[T]): ReadWriteAction[T] = {
+    val azureConfig = multiCloudWorkspaceConfig.azureConfig.getOrElse(throw new RawlsException("Config not present"))
+    // TODO these will come from the spend profile service in the future
+    val spendProfileId = azureConfig.spendProfileId
+    val azureTenantId = azureConfig.azureTenantId
+    val azureSubscriptionId = azureConfig.azureSubscriptionId
+    val azureResourceGroupId = azureConfig.azureResourceGroupId
+
+    traceDBIOWithParent("findByName", parentSpan)(_ => dataAccess.workspaceQuery.findByName(workspaceRequest.toWorkspaceName, Option(WorkspaceAttributeSpecs(all = false, List.empty[AttributeName])))) flatMap {
+      case Some(_) => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"Workspace ${workspaceRequest.namespace}/${workspaceRequest.name} already exists")))
+      case None => {
+        val workspaceId = UUID.randomUUID
+        for {
+          _ <- traceDBIOWithParent("createWorkspaceInWSM", parentSpan)(_ => DBIO.from(
+            Future(workspaceManagerDAO.createWorkspaceWithSpendProfile(workspaceId, workspaceRequest.name, spendProfileId, userInfo.accessToken))
+          ))
+          _ = logger.info(s"Creating cloud context in WSM [workspaceId = ${workspaceId}]")
+          cloudContextCreateResult <- traceDBIOWithParent("createCloudContext", parentSpan)(_ => DBIO.from(
+            Future(workspaceManagerDAO.createAzureWorkspaceCloudContext(workspaceId, azureTenantId, azureResourceGroupId, azureSubscriptionId, userInfo.accessToken))
+          ))
+          jobControlId = cloudContextCreateResult.getJobReport.getId
+          _ = logger.info(s"Polling on cloud context in WSM [jobControlId = ${jobControlId}]")
+          pollingResult <- traceDBIOWithParent("pollCloudContext", parentSpan)(_ => DBIO.from(
+            pollCloudContext(workspaceId, cloudContextCreateResult.getJobReport.getId, userInfo.accessToken)
+          ))
+          _ = logger.info(s"Creating workspace record [workspaceId = ${workspaceId}]")
+          savedWorkspace <- traceDBIOWithParent("saveNewWorkspaceToRawlsDB", parentSpan)(_ => {
+            createMultiCloudWorkspaceInDatabase(
```

Comment:
Can we limit the scope of the DB transaction to this call rather than including the WSM calls in the transaction?